### PR TITLE
cmd/explain_test: migrate collation test and add more tests

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: ref #31625

Problem Summary:
This PR migrate TestIssue27429 to explain test and add more tests related to the aggregate function.
And we found three bugs: https://github.com/pingcap/tidb/issues/31638 , https://github.com/pingcap/tidb/issues/31637 and https://github.com/pingcap/tidb/issues/31640

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: cmd/explaintest/r/collation_agg_func_enabled.result [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  └─StreamAgg	1.00	cop[tikv]		funcs:max(cast(collation_agg_func.tt.b, enum('a','B','c')))->Column#8
+    └─TableFullScan	10000.00	cop[tikv]	table:tt	keep order:false, stats:pseudo
+desc format='brief' select min(c) from tt;
+id	estRows	task	access object	operator info
+HashAgg	1.00	root		funcs:min(collation_agg_func.tt.c)->Column#6
+└─TableReader	10000.00	root		data:TableFullScan
+  └─TableFullScan	10000.00	cop[tikv]	table:tt	keep order:false, stats:pseudo
+select min(c) from tt;
+min(c)
+B
```
Comment (wjhuang2016) [inline cmd/explaintest/r/collation_agg_func_enabled.result:226]: It's wrong.

**Review by xiongjiwei — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**